.github/workflows/sonar.yaml

```diff
@@ -0,0 +1,23 @@
+name: Sonar Scan
+on: [push, pull_request_target]
+
+jobs:
+  sonar:
+    name: Sonar Scan
+    runs-on: ubuntu-latest
+    # always run on push events
+    # only run on pull_request_target event when pull request pulls from fork repository
+    if: >
+      github.event_name == 'push' || 
+      github.event_name == 'pull_request_target' && github.event.pull_request.head.repo.full_name != github.repository 
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - uses: sonarsource/sonarcloud-github-action@master
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+
```

Comment:
There's [no coverage information on this PR](https://sonarcloud.io/dashboard?branch=sdk-1854-github-actions&id=getyoti%3Ago), as there's no Go code. Would be good to check that the coverage works, but I guess the best way of doing that would to see how it runs on the next PR that modifies Go code